main/lsp/LSPTypechecker.cc

```diff
@@ -246,28 +246,40 @@ vector<core::FileRef> LSPTypechecker::runFastPath(LSPFileUpdates &updates, Worke
         fref.data(*gs).strictLevel = pipeline::decideStrictLevel(*gs, fref, config->opts);
 
         toTypecheck.emplace_back(fref);
+    }
+
+    UnorderedSet<core::FileRef> packageFiles;
 
+    for (auto fref : toTypecheck) {
         // Only need to re-run packager if we're going to delete constants and have to re-define
         // their visibility, which only happens if we're running incrementalNamer.
         if (shouldRunIncrementalNamer) {
```

Comment:
Maybe we should now invert the logic here:

```c++
if (shouldRunIncrementalNamer) {
    UnorderedSet<core::FileRef> packageFiles;

    for (auto fref : toTypecheck) {
        ...
    }

    // include them in `toTypecheck` if necessary
}
```

Reply:
ah yes, good call